hpcflow/archive/archive.py

```diff
@@ -70,7 +78,8 @@ class Archive(Base):
     name = Column(String(255))
     _path = Column('path', String(255))
     host = Column(String(255))
-    cloud_provider = Column(Enum(CloudProvider))
+    cloud_provider_type = Column(Enum(CloudProviderType))
+    cloud_provider = Column(PickleType)
```

Comment:
I don't think we need this to be a `PickleType`. The `CloudProvider` class does not really hold any instance attributes and so does not need a persistent representation within the database. Instead, the way to do this is by adding a method to `Archive` with the `sqlalchemy.orm.reconstructor` decorator; this method will be invoked when the object is instantiated from the database. So in this case, we can instantiate the `CloudProvider` instance within such a method.

Reply:
I think I have managed this.